Code/Source/RobotImporter/RobotImporterWidget.cpp

```diff
@@ -63,20 +76,23 @@ namespace ROS2
                     ReportError("User cancelled");
                     return;
                 }
-                RobotImporter::Import(
-                    { urdfPath.value(), prefabPath.value() },
-                    [this](const AZStd::string& message)
-                    {
-                        ReportInfo(message);
-                    },
-                    [this](const AZStd::string& message)
-                    {
-                        ReportError(message);
-                    });
+                m_robotImporter.Import({ urdfPath.value(), prefabPath.value() });
+                m_selectFileButton->setEnabled(false);
+                m_importerUpdateTimer->start(500);
             });
         setLayout(mainLayout);
     }
 
+    void RobotImporterWidget::ImporterTimerUpdate()
+    {
+        m_robotImporter.Update(
+            [this]()
+            {
+                m_importerUpdateTimer->stop();
```

Comment:
we could instead set this timer to single shot

Reply:
Not really. There is no time that we could set here to ensure that the import is done that will not be ridiculously high. I believe a regular timer that polls in a set interval is the only option. The interval time is up to discussion, however.